cmake/checks/directx.cpp

```diff
@@ -1,7 +1,6 @@
 #include <windows.h>
 
 #include <d3d11.h>
-#pragma comment (lib, "d3d11.lib")
```

Comment:
There is still several places in OpenCV codebase which use this approach (and they don't use LINK_LIBRARIES).
Fix should be complete and not introduce mess.

Try `cmake -DBUILD_EXAMPLES=ON ...`

Reply:
Understood. I have already built those. But I thought those will require a separate PR according to the guidelines. I shall add another commit here.

Reply:
Question. The samples/opengl/opengl_interop.cpp file contains pragma comment directive but it is not compiled for Windows platform because X11_FOUND is false. Would you like to take look at it?

Reply:
I have added the required changes. Please review.